cmd/juju/application/addunit.go

```diff
@@ -217,6 +252,28 @@ func (c *addUnitCommand) Run(ctx *cmd.Context) error {
 	}
 	defer apiclient.Close()
 
+	if c.unknownModel {
+		if err := c.validateArgs(); err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	modelType, err := c.ModelType()
```

Comment:
Same comment as for remove-unit - I think we pass model type to validate() method.
Also, can we be consistent with naming of the method between add and remove